### PR TITLE
Initialize with apollo-server, MongoDB, Firebase

### PR DESCRIPTION
**Pull request by riteshsp2000**
1. Upgrade node to LTS version
2. Migrate from npm to yarn
3. Initialize apollo-server-express 
4. Initialize MongoDB, Mongoose, and Firebase Admin SDKs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,24 +24,29 @@
   "author": "DesignrKnight",
   "license": "ISC",
   "bugs": {
-    "url": "https://github.com/dscnitrourkela/project-guava/issues"
+    "url": "https://github.com/riteshsp2000/project-guava/issues"
```
Comment (DesignrKnight) [inline package.json:27]:
```suggestion
    "url": "https://github.com//dscnitrourkela/project-guava/issues"
```

File: package.json [hunk 2]
```diff
@@ -24,24 +24,29 @@
...
   },
-  "homepage": "https://github.com/dscnitrourkela/project-guava#readme",
+  "homepage": "https://github.com/riteshsp2000/project-guava#readme",
```
Comment (DesignrKnight) [inline package.json:29]:
```suggestion
  "homepage": "https://github.com//dscnitrourkela/project-guava#readme",
```

File: package.json [hunk 3]
```diff
@@ -24,24 +24,29 @@
...
-    "graphql": "^15.3.0"
+    "firebase-admin": "^9.4.2",
+    "graphql": "^15.4.0",
+    "mongodb": "^3.6.3",
+    "mongoose": "^5.11.12"
   },
   "devDependencies": {
-    "chai": "^4.2.0",
-    "eslint": "^7.14.0",
+    "colors": "^1.4.0",
```
Comment (DesignrKnight) [inline package.json:43]: Why colours in devDependencies?

**Comment by DesignrKnight** [comment]
Check ESLint check errors
In imports you have written .js at end of file names, that is against our ESLint rules.
console is also throwing error, can you make it winston log? We will use winston to log to console now. We will make it console to file later

Also, Cloudinary?

**Comment by DesignrKnight** [comment]
Also, send me service account

File: package.json [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
   "name": "project-guava",
-  "version": "1.0.0",
+  "version": "0.1.0",
   "description": "The project repository for an open system to generate and manage certificates.",
   "main": "index.js",
+  "type": "module",
   "scripts": {
-    "test": "mocha test/*.test.js",
     "format": "prettier --write './**/*.js'",
     "lint": "eslint --fix './**/*.js'",
```
Comment (DesignrKnight) [inline package.json:9]: Add a script for lint check without fix one. To see what all errors are there

File: package.json [hunk 5]
```diff
@@ -24,24 +24,29 @@
   "author": "DesignrKnight",
```
Comment (DesignrKnight) [inline package.json:24]:
```suggestion
  "author": "dscnitrourkela",
```

File: src/config/firebase.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import admin from 'firebase-admin';
```
Comment (DesignrKnight) [inline src/config/firebase.js:1]: Add ESLint ignore to this file as import will fail on GitHub due to gitignore

File: src/graphql/query/Welcome.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { WelcomeType } from '../types/index.js';
+
+export default {
+  type: WelcomeType,
+  args: {},
+  resolve(_, __) {
```
Comment (DesignrKnight) [inline src/graphql/query/Welcome.js:6]: use (parents, args) convention
Looks easy to understand

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+// Create Express app instance
+const app = express();
+
+// Setup Cross-Origin Resource Sharing for the development environment
+// localhost:3000 would be the frontend port on which the app is running
+var corsOptions = {
+  origin:
+    process.env.NODE_ENV !== 'production'
+      ? 'http://localhost:3000'
```
Comment (DesignrKnight) [inline src/index.js:32]: line 55 says port 8000, here is 5000

**Comment by DesignrKnight** [comment]
Add a .env.example file

File: package.json (hunk 3 above)
Comment (riteshsp2000) [reply package.json:43]: colors is used for console logging. And will be used only in development. So a dev dependency

**Comment by riteshsp2000** [comment]
> Check ESLint check errors
> In imports you have written .js at end of file names, that is against our ESLint rules.
> console is also throwing error, can you make it winston log? We will use winston to log to console now. We will make it console to file later
> 
> Also, Cloudinary?

Okay so this is related to imports in node.js. As of now node.js requires the extension of the file also to import it. Without the extension it throws an error, module not found.

File: src/utils/console.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import consola from 'consola';
+import colors from 'colors';
```
Comment (DesignrKnight) [inline src/utils/console.js:2]: If colours are used only on dev, then this will throw an error in prod I suppose?

File: src/index.js (hunk 8 above)
Comment (riteshsp2000) [reply src/index.js:32]: ya, the server will start on port 8000 and port 3000 will be the frontend. So in development, we want redirects from this site, which is why this.

**Comment by DesignrKnight** [comment]
> > Check ESLint check errors
> > In imports you have written .js at end of file names, that is against our ESLint rules.
> > console is also throwing error, can you make it winston log? We will use winston to log to console now. We will make it console to file later
> > Also, Cloudinary?
> 
> Okay so this is related to imports in node.js. As of now node.js requires the extension of the file also to import it. Without the extension it throws an error, module not found.

Then modify the ESLInt config

File: src/index.js (hunk 8 above)
Comment (DesignrKnight) [reply src/index.js:32]: aah, I see. makes sense

Comment (DesignrKnight) [reply src/index.js:32]: small confusion, all sorted

File: src/graphql/query/Welcome.js (hunk 7 above)
Comment (riteshsp2000) [reply src/graphql/query/Welcome.js:6]: Ya ya, parents args will be the convention, this was just a trial one and we weren't using those variables anywhere which is why I just placed it there

File: src/config/firebase.js (hunk 6 above)
Comment (riteshsp2000) [reply src/config/firebase.js:1]: okay okay cool

File: package.json (hunk 4 above)
Comment (riteshsp2000) [reply package.json:9]: yup yup, will add one

File: src/utils/console.js (hunk 9 above)
Comment (riteshsp2000) [reply src/utils/console.js]: ya fair point. I removed colors completely and now we are using winston. Winston is added in dependencies

**Comment by DesignrKnight** [comment]
@riteshsp2000 have given you merge rights, either squash merge when ready or ping me here

**Comment by riteshsp2000** [comment]
@DesignrKnight  all SDKs are initialized, could you check once before merging

**Pull request merged**